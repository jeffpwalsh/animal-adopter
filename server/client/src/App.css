* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  text-align: center;
}

.helpButton {
  border-radius: 50%;
  height: 50px;
  width: 50px;
  color: black;
  border: none;
}

.round-img {
  border-radius: 50%;
  width: 250px;
  height: 250px;
  margin-bottom: 20px;
}

ul,
a {
  list-style: none;
  text-decoration: none;
}

body {
  background: url('./img/showcase2.jpg') fixed no-repeat center center/cover;
  position: relative;
  height: 100vh;
}

/* navbar */
.navbar {
  display: flex;
  justify-content: space-between;
  position: fixed;
  width: 100%;
  top: 5px;
  opacity: 0.9;
}
nav.navbar {
  background-color: #555555;
}
.navbar ul {
  display: flex;
}
a.navbarHeading {
  color: #fff;
}

a.navbarHeading:hover {
  text-decoration: none;
  color: #dc3545;
}

.navbar a {
  color: #fff;
  padding: 0.45rem;
  margin: 0 0.25rem;
}

.heart {
  color: #dc3545;
}

/* Landing Page */
.landing {
  position: relative;
  height: 100vh;
}

.landing-inner {
  color: #fff;
  height: 100%;
  width: 80%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dark-overlay {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.signup {
  color: #dc3545;
}

small {
  font-size: 18px;
  color: #f3f3f3;
}

p {
  font-weight: lighter;
  font-size: 18px;
}

.btn {
  margin: 0.25em;
}

/* login */

.login {
  position: relative;
  height: 50vh;
  margin-top: 25%;
}

.login-inner {
  color: #fff;
  height: 100%;
  width: 50%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dark-overlay2 {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}


/* register */

.register {
  position: relative;
  height: 50vh;
  margin-top: 25%;
}

.register-inner {
  color: #fff;
  height: 100%;
  width: 50%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dark-overlay3 {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

/* dashboard */

.dashboard {
  position: relative;
  height: 70vh;
  margin-top: 25%;
}

.dashboard-inner {
  color: #fff;
  height: 100%;
  width: 60%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dark-overlay-dashboard {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

/* profile edit  */

.profileEdit {
  position: relative;
  height: 90vh;
  margin-top: 25%;
}

.profileEdit-inner {
  color: #fff;
  height: 100%;
  width: 80%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dark-overlay-profileEdit {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.form {
  display: block;
  margin-top: 0.3rem;
  color: #888;
}

.form input,
.form input,
.form input,
.form input,
.form select,
.form textarea {
  display: block;
  width: 100%;
  padding: 0.4rem;
  font-size: 1.2rem;
}

/* Info */
.infoAdd {
  position: relative;
  height: 90vh;
  margin-top: 25%;
}

.infoAdd-inner {
  color: #fff;
  height: 100%;
  width: 80%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dark-overlay-infoAdd {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

/* profile */
.profileAll {
  position: relative;
  height: 100vh;
  margin-top: 25%;
}

.inner-profileAll {
  color: #fff;
  height: 100%;
  width: 80%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dark-overlay-profileAll {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

/* adoption profiles */

.adoptionProfile {
  position: relative;
  height: 70vh;

  margin-top: 10%;
}

.inner-adoptionProfile {
  color: #fff;
  height: 100%;
  width: 80%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dark-overlay-adoptionProfile {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

span,
.highlight {
  color: #ffc107;
}

.displayProfiles {
  text-align: center;
  color: #f3f3f3;
  padding-top: 20%;
}

.displayProfiles h1 {
  background-color: #dc3545;
  width: max-content;
  margin: auto;
  padding: 10px;
  border-radius: 5px;
}

/* single profile display */

.profileSingle {
  position: relative;
  height: 125vh;
  margin-top: 25%;
}

.inner-profileSingle {
  color: #fff;
  height: 100%;
  width: 80%;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.dark-overlay-profileSingle {
  background-color: rgba(0, 0, 0, 0.7);
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.profile-top,
.profile-about {
  margin-bottom: 50px;
}

.spacer {
  margin-top: 50px;
}
.spacer-bottom {

  margin-bottom: 50px;
}

.characteristics {

font-weight: lighter;
font-size: 20px; 

}
