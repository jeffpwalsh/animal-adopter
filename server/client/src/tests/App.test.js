import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';
import renderer from 'react-test-renderer';

test('Snapshot Test App', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


