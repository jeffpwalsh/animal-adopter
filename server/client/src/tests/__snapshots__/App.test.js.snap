// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test App 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg"
  >
    <h1>
      <a
        className="navbarHeading"
        href="/"
        onClick={[Function]}
      >
        Animal
        <span
          className="heart"
        >
          ‚ù§
        </span>
        Adopter
      </a>
    </h1>
    <button
      className="helpButton btn btn-warning"
      onClick={[Function]}
    >
      Help
    </button>
  </nav>,
  <section
    className="landing"
  >
    <div
      className="dark-overlay"
    >
      <div
        className="landing-inner"
      >
        <h1
          className="x-large"
        >
          Animal Adopter
        </h1>
        <p>
          Create a animal profile to find our friends a home
        </p>
        <div
          className="buttons"
        >
          <a
            className="btn btn-danger"
            href="/register"
            onClick={[Function]}
          >
            Sign Up
          </a>
          <a
            className="btn btn-outline-danger"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </section>,
]
`;
